src/flow_solver/flow_solver_cases/periodic_turbulence.h

```diff
@@ -141,6 +145,21 @@ class PeriodicTurbulence : public PeriodicCubeFlow<dim,nstate>
     /// Maximum local wave speed (i.e. convective eigenvalue)
     double maximum_local_wave_speed;
 
+    /// Previous time
+    double previous_time = 0;
+
+    /// Target dt for calculating relaxation parameter
+    double dt_target = 1;
+
+    /// Numerical entropy at previous timestep
+    double previous_numerical_entropy = 0;
+
+    /// Cumulative change in numerical entropy
+    double cumulative_numerical_entropy_change_FRcorrected = 0;
+
+    /// Numerical entropy at initial time
+    double initial_numerical_entropy_abs = 0;
+
```

Comment:
i think that maybe a class derived from `PeriodicTurbulence` might be more appropriate since you're introducing all these numerical entropy related things and its separate from Turbulence, maybe something like `PeriodicTurbulenceEntropyTests`? It would also allow you to replace the if statements by polymorphism

Reply:
I would prefer to keep it in one class as I think it's more consistent to have the flow solver case chosen only by `flow_type`. If I were to make an additional class, I would break this pattern. 

As a compromise, I have reorganized the numerical entropy calculation into a separate function to keep the code a bit more tidy.

Reply:
That might be possible, but I would need time to think about how best to organize it. I may implement that the next time that I am making significant changes to this part of the code.

Reply:
Per our discussion in the lab, I think that reorganizing the `flow_solver_case` structure for entropy tests is a good idea. It won't get done on this PR, but is on my to-do list for the next time I need to work on this part of the code.

Reply:
Sounds good, just reference this comment in a new issue and you can resolve this

Reply:
https://github.com/dougshidong/PHiLiP/pull/195/files/c07024424c16a0aece8bda24603add99962cb3f4#r1523380835